### PR TITLE
Enip rust 3958 v4

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3958
https://redmine.openinfosecfoundation.org/issues/6304

Describe changes:
- fix stats not having `_tcp` prefix when protocol is `detection-only`
- convert enip parser to rust

Alon the way, also
- enip_command keyword accepts now string enumeration as values.
- transactions are now bidirectional
- there is a enip logger
- gap support is improved with probing for resync
- SEQUENCE_ADDR_ITEM value is fixed to 0x8002 instead of 0xB002
- frames
- events

#9848  CI should be greener + enip.status keyword + enip identity parsing and logging

Draft as is this is not complete :
- parse more, log more, add more keywords (what is logged in enip identity)...

### Provide values to any of the below to override the defaults.

```
SV_BRANCH=pr/1485
```
https://github.com/OISF/suricata-verify/pull/1485

### CODE REVIEW

File: rust/src/enip/detect.rs [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+    }
+}
+
+fn enip_parse_command(i: &str) -> IResult<&str, u16> {
+    let (i, v) = alt((enip_detect_parse_u16, enip_parse_command_string))(i)?;
+    return Ok((i, v));
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn rs_enip_parse_command(
```
Comment (jasonish) [inline rust/src/enip/detect.rs:72]: Public C API, so `SCEnipParseCommand`?

Comment (jasonish) [reply rust/src/enip/detect.rs:72]: Basically apply this to any no_mange pub extern C fn.

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 184 | 174 | 94.57% | 

Pipeline 16661

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 16664

File: rust/src/enip/detect.rs (hunk 1 above)
Comment (victorjulien) [reply rust/src/enip/detect.rs:72]: We should probably have a (tracking) ticket to clean this up for any existing code. Plus see if we can have a CI check?

**Comment by victorjulien** [comment]
"SEQUENCE_ADDR_ITEM value is fixed to 0x8002 instead of 0xB002" do we need this as a fix for 6/7 too?

**Comment by catenacyber** [comment]
> "SEQUENCE_ADDR_ITEM value is fixed to 0x8002 instead of 0xB002" do we need this as a fix for 6/7 too?

I guess so, but I have no pcap for test, just https://github.com/wireshark/wireshark/blob/136ca4287d66b84b2bbd46d616530abb458ddfdc/epan/dissectors/packet-enip.c#L89

Also, this PR does not pretend to handle `CIP_SET_ATTR_LIST` (when current version does it wrong for more than one attribute)

**Pull request closed without merging**